aesd-char-driver/aesd-circular-buffer.c

```diff
@@ -44,9 +84,26 @@ struct aesd_buffer_entry *aesd_circular_buffer_find_entry_offset_for_fpos(struct
 */
 void aesd_circular_buffer_add_entry(struct aesd_circular_buffer *buffer, const struct aesd_buffer_entry *add_entry)
 {
-    /**
-    * TODO: implement per description 
-    */
+    struct aesd_buffer_entry* aesd_buff_entry = (struct aesd_buffer_entry*) malloc(sizeof(struct aesd_buffer_entry));
```

Comment:
@rajatchaple you could just use the stack here to avoid memory allocation, but also I'm not sure this is needed.  Where is it freed?